test/test_python_dispatch.py

```diff
@@ -497,5 +497,187 @@ def wrap(e):
         self.assertEqual(x.grad, y)
         self.assertEqual(y.grad, x)
 
+
+class FunctionalTensor(torch.Tensor):
+
+    elem: torch.Tensor
+
+    __slots__ = ['elem', 'elem_']
+
+    @staticmethod
+    def __new__(cls, elem, *args, **kwargs):
+        # r = torch.Tensor._make_wrapper_subclass(cls, elem)
+        r = torch.Tensor._make_wrapper_subclass(
+            cls, elem.size(), elem.stride(),
+            dtype=elem.dtype, layout=elem.layout,
+            device='meta', requires_grad=elem.requires_grad
+        )
+        if elem._is_functional_tensor():
+            r.elem_ = elem
+            r.elem = r.elem_
+        else:
+            r.elem_ = elem
+            r.elem = r.elem_._to_functional_tensor()
+        return r
+
+    def __repr__(self):
+        assert self.elem._is_functional_tensor()
+        self.elem._sync()
+        return f"FunctionalTensor({self.elem._from_functional_tensor()})"
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
```

Comment:
I'm not sure why this `__torch_dispatch__` is necessary at all. Don't we have registered kernels for the Functional dispatch key anyway; why isn't this simply a matter of just doing ordinary dispatch?